dhis-2/dhis-support/dhis-support-system/src/test/java/org/hisp/dhis/system/util/ValidationUtilsTest.java

```diff
@@ -620,4 +624,33 @@ void testIsValidLetter()
         List<String> invalid = List.of( "1", "", "aa", "=", "," );
         invalid.forEach( value -> assertFalse( isValidLetter( value ) ) );
     }
+
+    @Test
+    void shouldPassValidationWhenRenderTypeIsCanvasAndValueTypeIsImageAndNoOptionSet()
+    {
+        // Both ProgramStageDataElement and ProgramTrackedEntityAttribute should be supported.
+        assertTrue( validateRenderingType( ProgramStageDataElement.class, IMAGE, false, ValueTypeRenderingType.CANVAS ),
+            "ProgramStageDataElement of type IMAGE fails renderingType validation for CANVAS" );
+        assertTrue(
+            validateRenderingType( ProgramTrackedEntityAttribute.class, IMAGE, false, ValueTypeRenderingType.CANVAS ),
+            "ProgramTrackedEntityAttribute of type IMAGE fails renderingType validation for CANVAS" );
+    }
+
+    @Test
+    void shouldFailValidationWhenRenderTypeIsCanvasAndValueTypeIsNotImageOrOptionSet()
```

Comment:
I am not really familiar with that but I cannot understand clearly from the test case what it being tested.
Maybe this test can be divided into 2 different ones, so from the name it will be more clear?

From the name I don't understand why the third and fourth assertions are false